Update sample chapter (#65)
oreillymedia/orm_book_samples: asciidoc_only/sample_chapter.asciidoc

----
====

> For either type of listing, you have the option of specifying the code language displayed; see the Code section of the Atlas user docs here -  https://docs.atlas.oreilly.com/writing_in_asciidoc.html for more details.

Reviewer: ```suggestion
For either type of listing, you have the option of specifying the code language displayed; see the Code section of the https://docs.atlas.oreilly.com/writing_in_asciidoc.html[Atlas user docs] for more details.
```

To insert a cross-reference, first add a block ID for the element to which you're referring. The block ID is a unique identifier enclosed in double square brackets. Place it on a single line above the block element (figure, table, section, etc.).

Here's a cross-reference to <<example_code_block>>. Here's one to <<example_table>>. Finally, here's one to <<sample_chapter_unique_id>>.

For the book to be valid, ids must be unique across the entire book, have no spaces, and not start with a number.

O'Reilly house style calls for every formal figure, formal table, and formal code example to be preceded by a specific in-text reference. Formal figures and formal tables are created when you add captions to these elements.

To include hyperlinks to external sources, add the full URL followed by the anchor text enclosed in single square brackets. This, for example, is a link to the https://www.oreilly.com/[O'Reilly home page]. The bracketed text will become a clickable link in web versions. In print versions, it will appear in the text, followed by the actual URL in parentheses.

> === Creating a Book Build in Atlas 

Reviewer: ```suggestion
=== Creating a Book Build in Atlas 

```

(Add an extra space to ensure proper conversion/this is generally good practice for authors)

> A helpful feature of Atlas is to create a book build in real time. This build shows the interior design of the book in PDF format, which allows you to see how elements will render on the print page. 

Reviewer: ```suggestion
A helpful feature of Atlas is to create a book build in real time. The PDF build shows the interior design of the book in PDF format, which allows you to see how elements will render on the print page, while the EPUB build generates an epub. 
```

> First, go to the Dashboard tab of your book project. Then, look to the right panel, click the "PDF" box, and hit "Build". Atlas will take a moment to generate the pages. A "Download" button will appear - click this, and a PDF will open in your web browseer. This PDF may then be saved locally. 

Reviewer: ```suggestion
To build a PDF, go to the Dashboard tab of your book project. Then, click the "PDF" checkbox box under "Builds", and hit "Build" Atlas will take a moment to generate the pages. A "Download" button will appear. Click this to open the PDF.
```

(The open-in-browser thing works only in Chrome, I think; we can be less explicit here -- we ought to assume our authors will be familiar with viewing/downloading PDFs)

[TIP]
====
If the Build button on the configure tab is grayed out, simply refresh the browser. The Build button should then become active.
====

> ==== Adding a Chapter to the Book Build 

Reviewer: Question: would it be worth noting that the same thing can be done in the _atlas.json_ file? Typically, I think authors who write in asciidoc/use atlas will be comfortable editing the json directly also, as an option.

Author: Good idea, added this in a Tip element

> In order to add a new chapter in Atlas, you'll need to make an adjustment to the Cofnigure tab in Atlas. 

Reviewer: ```suggestion
In order to add a new chapter to your book in Atlas, you'll need to make an adjustment to the Configure tab in Atlas. 
```

Author: Fixed

> In order to add a new chapter in Atlas, you'll need to make an adjustment to the Cofnigure tab in Atlas. 

Reviewer: ```suggestion
In order to add a new chapter in Atlas, you'll need to add the file using the Configure tab in your book's repository page. 
```

. Move to the Configure tab in the book project in Atlas. 

> . Scroll down to the middle section of the subsequent page under "Files". Find the chapter you'd like to add, and click the + symbol.  

Reviewer: ```suggestion
. Scroll to the middle section of the page under "Files." Find the chapter you'd like to add and click the + symbol.  
```

Author: Changed according to the suggestion.

> . Scroll down to the middle section of the subsequent page under "Files". Find the chapter you'd like to add, and click the + symbol.  

Reviewer: ```suggestion
. Scroll down to the middle section of the subsequent page under "Files." Find the chapter you'd like to add, and click the + symbol.  
```

. This will move the chapter to the "Build Contents" section on the right side of the page. 
.  Use the three stacked lines to reorder the chapters as desired. 

> . Once your contents list looks correct, scroll down the page to click the "Save Settings" button. 

Reviewer: ```suggestion
. Once your contents list looks correct, scroll down the page to click the "Save Settings."
```

Author: Fixed.

> .  Go back to the Dashboard tab, and repeat the steps to create a Book Build as discussed in the previous section. 

Reviewer: ```suggestion
.  Go back to the Dashboard and run a new book build as discussed in the previous section to verify your changes. 
```

Author: Changed according to the suggestion.

O'Reilly house style calls for every formal figure, formal table, and formal code example to be preceded by a specific in-text reference. Formal figures and formal tables are created when you add captions to these elements.

To include hyperlinks to external sources, add the full URL followed by the anchor text enclosed in single square brackets. This, for example, is a link to the https://www.oreilly.com/[O'Reilly home page]. The bracketed text will become a clickable link in web versions. In print versions, it will appear in the text, followed by the actual URL in parentheses.

=== Creating a Book Build in Atlas 
A helpful feature of Atlas is to create a book build in real time. This build shows the interior design of the book in PDF format, which allows you to see how elements will render on the print page. 

First, go to the Dashboard tab of your book project. Then, look to the right panel, click the "PDF" box, and hit "Build". Atlas will take a moment to generate the pages. A "Download" button will appear - click this, and a PDF will open in your web browser. This PDF may then be saved locally. 

[TIP]
====

> If the Build button on the configure tab is grayed out, simply refresh the browser. The Build button should then become active.

Reviewer 2: I think we should remove "on the configure tab".

Author: Good catch! I fixed this.
